src/oci/runc.rs

```diff
@@ -0,0 +1,299 @@
+//! Interface to [runc][0], a OCI compliant cli for spawning and running containers.
+//!
+//! [0]: https://github.com/opencontainers/runc
+
+
+#![allow(dead_code)]
+
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use derive_builder::Builder;
+use dyn_clone::{clone_trait_object, DynClone};
+use getset::{Getters, Setters};
+use std::{
+    fmt::{self, Debug},
+    path::{Path, PathBuf},
+    process::Output,
+    string::ToString,
+};
+use strum::AsRefStr;
+use tokio::process::Command;
+use std::ffi::OsStr;
+
+const RUNC_SUB_CMD_CHECKPOINT : &str = "checkpoint";
```

Comment:
this seems bad from my point of view, I could make it it's own enum like the args and just remove `write!(f, "--")?;` I just don't know if that buys us anything in the concept of trying to make a function call for each sub command to runc to make this struct easy to use.